crates/test-programs/wasi-tests/src/utils.rs

```diff
@@ -52,3 +52,32 @@ pub unsafe fn cleanup_file(dir_fd: wasi_unstable::Fd, file_name: &str) {
 pub unsafe fn close_fd(fd: wasi_unstable::Fd) {
     assert!(wasi_unstable::fd_close(fd).is_ok(), "closing a file");
 }
+
+// Returns: (rights_base, rights_inheriting)
+pub unsafe fn fd_get_rights(fd: wasi_unstable::Fd) -> (wasi_unstable::Rights, wasi_unstable::Rights) {
+    let mut fdstat: wasi_unstable::FdStat = std::mem::zeroed();
+    let status = wasi_fd_fdstat_get(fd, &mut fdstat);
+    assert_eq!(
+        status,
+        wasi_unstable::raw::__WASI_ESUCCESS,
+        "calling fd_fdstat_get"
+    );
+
+    (fdstat.fs_rights_base, fdstat.fs_rights_inheriting)
+}
+
+pub unsafe fn drop_rights(
```

Comment:
Likewise for this function, although a utility function that does the same thing using `wasi` would make sense.